roles/metallb/tasks/main.yml

```
@@ -0,0 +1,119 @@
+# MIT License
+#
+# (C) Copyright 2023 Hewlett Packard Enterprise Development LP
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+---
+- name: Copy MetalLB manifest
+  ansible.builtin.template:
+    src: "metallb-crds.yaml.j2"
+    dest: "{{ k3s_server_location }}/server/manifests/metallb-crds.yaml"
+    owner: root
+    group: root
+    mode: '0644'
+
+- name: Create temp manifests directory for metallb configuration
+  ansible.builtin.file:
+    path: "/tmp/k3s/metallb"
+    state: directory
+    owner: "{{ ansible_user_id }}"
+    mode: '0755'
+
+- name: Copy MetalLB CRs manifest
+  ansible.builtin.template:
+    src: "metallb-crs.yaml.j2"
+    dest: "/tmp/k3s/metallb/metallb-crs.yaml"
+    owner: root
+    group: root
+    mode: '0644'
+
+- name: Pause for MetalLB resources to load
+  ansible.builtin.pause:
+    seconds: 10
+
+- name: Test metallb-system namespace
+  ansible.builtin.command: >-
+    /usr/local/bin/k3s kubectl -n metallb-system
+  changed_when: false
+
+- name: Wait for MetalLB resources
+  ansible.builtin.command: >-
+    /usr/local/bin/k3s kubectl wait {{ item.resource }}
+    --namespace='metallb-system'
+    {% if item.name | default(False) -%}{{ item.name }}{%- endif %}
+    {% if item.selector | default(False) -%}--selector='{{ item.selector }}'{%- endif %}
+    {% if item.condition | default(False) -%}{{ item.condition }}{%- endif %}
+    --timeout='{{ metal_lb_available_timeout }}'
+  changed_when: false
+  run_once: true
+  with_items:
```

This got clobbered, but please use `loop` instead (even if you copied this from somewhere else).
```suggestion
  loop:
```